src/lists/list.jsx

```diff
@@ -85,16 +88,45 @@ const List = React.createClass({
       subheaderElement = <div style={mergedSubheaderStyles}>{subheader}</div>;
     }
 
+    let listItems;
+    if (this.props.selectedLink) {
+      listItems = React.Children.map(children, (child) => {
+        if (child.type.displayName === "ListItem") {
+          return React.cloneElement(child, {
+              key: child.props.index,
+              selected: this._getSelected(child.props.index),
+              updateSelected: this._updateSelectedIndex,
+            }
+          );
+        }
+        else {
+          return child;
+        }
+      });
+    }
+    else {
+      listItems = children;
+    }
+
     return (
       <Paper
         {...other}
         style={this.mergeStyles(styles.root, style)}
         zDepth={zDepth}>
         {subheaderElement}
-        {children}
+        {listItems}
       </Paper>
     );
   },
+
+  _getSelected(index) {
+    return this.props.selectedLink.value === index;
+  },
+
+  _updateSelectedIndex(index) {
```

Comment:
Actually, I would keep this method to keep the same reference for the `updateSelected` property.
This can save use some rerendering when using the purerender mixin.